src/Servers/Kestrel/Core/src/Internal/HttpsConnectionMiddleware.cs

```diff
@@ -69,33 +71,58 @@ public HttpsConnectionAdapter(HttpsConnectionAdapterOptions options, ILoggerFact
             }
 
             _options = options;
-            _logger = loggerFactory?.CreateLogger<HttpsConnectionAdapter>() ?? (ILogger)NullLogger.Instance;
+            _logger = loggerFactory?.CreateLogger<HttpsConnectionMiddleware>();
         }
-
-        public bool IsHttps => true;
-
-        public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext context)
+        public Task OnConnectionAsync(ConnectionContext context)
         {
-            // Don't trust SslStream not to block.
             return Task.Run(() => InnerOnConnectionAsync(context));
         }
 
-        private async Task<IAdaptedConnection> InnerOnConnectionAsync(ConnectionAdapterContext context)
+        private async Task InnerOnConnectionAsync(ConnectionContext context)
         {
             SslStream sslStream;
             bool certificateRequired;
             var feature = new Core.Internal.TlsConnectionFeature();
             context.Features.Set<ITlsConnectionFeature>(feature);
             context.Features.Set<ITlsHandshakeFeature>(feature);
 
+            // TODO: Handle the cases where this can be null
+            var memoryPoolFeature = context.Features.Get<IMemoryPoolFeature>();
+
+            var inputPipeOptions = new PipeOptions
+            (
+                pool: memoryPoolFeature.MemoryPool,
+                readerScheduler: _options.Scheduler,
+                writerScheduler: PipeScheduler.Inline,
+                pauseWriterThreshold: _options.MaxInputBufferSize ?? 0,
+                resumeWriterThreshold: _options.MaxInputBufferSize / 2 ?? 0,
+                useSynchronizationContext: false,
+                minimumSegmentSize: memoryPoolFeature.MemoryPool.GetMinimumSegmentSize()
+            );
+
+            var outputPipeOptions = new PipeOptions
+            (
+                pool: memoryPoolFeature.MemoryPool,
+                readerScheduler: PipeScheduler.Inline,
+                writerScheduler: PipeScheduler.Inline,
+                pauseWriterThreshold: _options.MaxOutputBufferSize ?? 0,
+                resumeWriterThreshold: _options.MaxOutputBufferSize / 2 ?? 0,
+                useSynchronizationContext: false,
+                minimumSegmentSize: memoryPoolFeature.MemoryPool.GetMinimumSegmentSize()
+            );
+
+            // TODO: eventually make SslDuplexStream : Stream, IDuplexPipe to avoid RawStream allocation and pipe allocations
```

Comment:
This TODO on the other hand is fine. This seems like a fair bit of work.

Reply:
I'll tackle them across the code base in a separate PR. 

PS: https://github.com/aspnet/AspNetCore/blob/0648edf7c2c481309272697c0e0b60e47696f9f7/src/Servers/Kestrel/Transport.Libuv/src/Internal/Networking/UvStreamHandle.cs#L47